lib/base-package-manager.ts

```diff
@@ -0,0 +1,94 @@
+import * as child_process from 'child_process';
+
+export class BasePackageManager {
+	constructor(private packageManager: string) { }
+
+	protected getNpmExecutableName(isWindows: boolean): string {
```

Comment:
No need to provide `isWindows` as param. You can determinate it inside the method. Just need to inject `$hostInfo`.
```
constructor(private packageManager: string,
    private $hostInfo: IHostInfo){ }
```
```
protected getNpmExecutableName(): string {
    const isWindows = this.$hostInfo.isWindows;
}
```

You'll need to change this https://github.com/NativeScript/nativescript-cli/pull/4050/files#diff-5d0fca739bb2fbd7cab602dbeaa30b89R17 as well

```
super('npm', $hostInfo);
```

This also should be changed https://github.com/NativeScript/nativescript-cli/pull/4050/files#diff-5d0fca739bb2fbd7cab602dbeaa30b89L11
```
constructor($hostInfo: IHostInfo, ...) { }
```
NOTE: No `private` before `$hostInfo`.